src/prefect/new_task_engine.py

```diff
@@ -502,6 +508,7 @@ def run_task(
         parameters=parameters,
         wait_for=wait_for,
         return_type=return_type,
+        dependencies=dependencies,
```

Comment:
I added this `dependencies` kwarg to permit providing a list of `TaskRunInput`s directly for dependency tracking since we are resolving top level futures for `Task.map`. This will also be useful for the `PrefectDistributedClient` that will also handle dependency tracking responsibilities.